sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -488,6 +489,30 @@ class FileIndexSuite extends SharedSparkSession {
     val fileIndex = new TestInMemoryFileIndex(spark, path)
     assert(fileIndex.leafFileStatuses.toSeq == statuses)
   }
+
+  test("expire FileStatusCache if TTL is configured") {
+    val previousValue = SQLConf.get.getConf(StaticSQLConf.METADATA_CACHE_TTL)
+    try {
+      SQLConf.get.setConf(StaticSQLConf.METADATA_CACHE_TTL, 1L)
```

Comment:
we can use `withSQLConf`

Reply:
unfortunately `withSQLConf` doesn't work with `StaticSQLConf`

Reply:
ah i see, it's kind of a workaround to set static conf at runtime. Maybe we should add a comment to speak it out.

Reply:
added a comment in https://github.com/apache/spark/pull/28852/commits/3e761dcd790b9c30e5cee7bffe916dfc2c82b7a5